### PR TITLE
REFACTOR-#5330: remove `BaseIO._read`

### LINKED ISSUE #5330: Fix code scanning alert - Conflicting attributes in base classes
Tracking issue for:
- [ ] https://github.com/modin-project/modin/security/code-scanning/342

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

We already have `_read` methods in the dispatcher hierarchy. Since there is no need to make such a method also in the IO hierarchy, we can remove it and this will help us stop figuring out which method from the hierarchy was called.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5330 
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
Blocked on https://github.com/modin-project/modin/pull/5328

**Review by mvashishtha — APPROVED** [review]

**Comment by mvashishtha** [comment]
@modin-project/modin-hdk need approval please. @vnlitvinov ?

**Review by Garra1980 — APPROVED** [review]

**Pull request merged**
